### DISCUSSION TITLE
HELP ASAP

### DISCUSSION THREAD

**Discussion by Jakewinkel91** [theyosh/TerrariumPI#703] [Q&A] [answered]
I am having problems running TerrariumPi on my pi 3b+, i had it running fine for the past two months until today, when I decided to clean up my wiring.  After changing the wiring the pi booted but the server wouldn't be visible. so I reinstalled everything on the SD card.  the problem I am having is now I get a failed error when the pi starts up "failed to start terrariumpi.service.  if I try to start the service manually start the service  it says failed because the control process exited with error code!

Help would be appreciated, my orchids will thank you!
Jake

**Comment by theyosh** [comment]
This should give some more info about errors: https://theyosh.github.io/TerrariumPI/faq/how-to-debug-terrariumpi/

**Comment by Jakewinkel91** [comment]
I think i'm going to limp my terrarium until pay day when I can give my
left arm for a Pi4..... I have had a bunch of issues like not being able to
get irf520s to work, the camera wont work, the usb sound card wont work....
Hopefully a pi 4 will put me in a better position!

**Reply by theyosh** [reply]
New Raspberrys are scarce.. Chip shortage. But what if you just run the debug option... maybe it is a small fix...

**Reply by Jakewinkel91** [reply]
pi@raspberrypi:~ $ sudo service terrariumpi stop
pi@raspberrypi:~ $ cd /home/pi/TerrariumPI/
pi@raspberrypi:~/TerrariumPI $ source venv/bin/activate
-bash: venv/bin/activate: No such file or directory
pi@raspberrypi:~/TerrariumPI $

**Reply by theyosh** [reply]
This looks like a broken installation. Not sure why. More info about the installation: https://theyosh.github.io/TerrariumPI/install/

Make sure you have used the (old) Legacy Raspberry PI Lite image (Buster).

Then do the installation once more after you have checked out the code. It looks like the installation is not complete. So rerun the installer should fix that.

**Reply by Jakewinkel91** [reply]
I don't know at this point, I will try again, wiped my sd card a few times yesterday and reinstalled! I will try again today!

**Comment by theyosh** [comment]
If you did a reinstall, please make sure you use the old (legacy) raspberry pi lite image (Buster). The default image will cause issues with the camera.

**Comment by Jakewinkel91** [comment]
So i wiped the SC card, installed the legacy buster image, followed the steps on the install page. still get the same error
![PXL_20220515_174107344](https://user-images.githubusercontent.com/92556100/168486508-1f1d868c-3c71-4013-ad75-f3eeae9fc6d6.jpg)
!

**Comment by theyosh** [comment, marked as answer]
Ok, found the problem. There was an error in the installation script. Sorry for that.

So do a `git pull` in the TerrariumPI folder and rerun the installer: `sudo ./install.sh` and then it should work.

**Reply by Jakewinkel91** [reply]
That worked, Irf520 next? I have 17v going in and 7v coming out? I have tried actual pwm pins and non pwm pins with the same results, changing the frequency doesn't change anything!
